src/time.rs

```diff
@@ -0,0 +1,118 @@
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly"))]
+        CLOCK_MONOTONIC_PRECISE,
+        #[cfg(any(target_os = "fuchsia",
+                  target_os = "linux",
+                  target_os = "android",
+                  target_os = "emscripten"))]
+        CLOCK_MONOTONIC_RAW,
+        #[cfg(any(target_os = "fuchsia",
+                  target_env = "uclibc",
+                  target_os = "macos",
+                  target_os = "ios",
+                  target_os = "freebsd",
+                  target_os = "dragonfly",
+                  target_os = "linux",
+                  target_os = "android",
+                  target_os = "emscripten"))]
+        CLOCK_PROCESS_CPUTIME_ID,
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly"))]
+        CLOCK_PROF,
+        CLOCK_REALTIME,
+        #[cfg(any(target_os = "fuchsia",
+                  target_os = "linux",
+                  target_os = "android",
+                  target_os = "emscripten"))]
+        CLOCK_REALTIME_ALARM,
+        #[cfg(any(target_os = "fuchsia",
+                  target_os = "linux",
+                  target_os = "android",
+                  target_os = "emscripten"))]
+        CLOCK_REALTIME_COARSE,
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly"))]
+        CLOCK_REALTIME_FAST,
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly"))]
+        CLOCK_REALTIME_PRECISE,
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly"))]
+        CLOCK_SECOND,
+        #[cfg(any(target_os = "fuchsia",
+                  target_os = "emscripten",
+                  all(target_os = "linux", target_env = "musl")))]
+        CLOCK_SGI_CYCLE,
+        #[cfg(any(target_os = "fuchsia",
+                  target_os = "emscripten",
+                  all(target_os = "linux", target_env = "musl")))]
+        CLOCK_TAI,
+        #[cfg(any(target_os = "fuchsia",
+                  target_os = "ios",
+                  target_os = "macos",
+                  target_os = "freebsd",
+                  target_os = "dragonfly",
+                  target_os = "linux",
+                  target_os = "android",
+                  target_os = "emscripten",
+                  target_env = "uclibc"))]
+        CLOCK_THREAD_CPUTIME_ID,
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly"))]
+        CLOCK_UPTIME,
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly"))]
+        CLOCK_UPTIME_FAST,
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly"))]
+        CLOCK_UPTIME_PRECISE,
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly"))]
+        CLOCK_VIRTUAL,
+    }
+}
+
+pub fn clock_getres(clk_id: ClockId) -> Result<TimeSpec> {
+    let mut c_time = libc::timespec {tv_sec: 0, tv_nsec: 0};
```

Comment:
Please use `MaybeUninit` instead of zero-initializing.  That will cause the test to fail for now, but they'll pass as soon as #1096 is fixed.

Reply:
`MaybeUninit` is the correct thing to do here, not zero-initialization.